ddtrace/opentelemetry/span_test.go

```diff
@@ -25,40 +26,76 @@ import (
 	"go.opentelemetry.io/otel"
 	"go.opentelemetry.io/otel/attribute"
 	"go.opentelemetry.io/otel/codes"
-	oteltrace "go.opentelemetry.io/otel/trace"
 )
 
-func mockTracerProvider(t *testing.T, opts ...tracer.StartOption) (tp *TracerProvider, payloads chan string, cleanup func()) {
-	payloads = make(chan string)
-	s, c := httpmem.ServerAndClient(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+func TestMain(m *testing.M) {
```

Comment:
Won't having a single server & client used by every test like this increase cross-test interference? For example, in [this failure](https://github.com/DataDog/dd-trace-go/actions/runs/7557030261/job/20575390915) we see subsequent tests fail with an unexpected payload, when that's probably bleeding over from another test hitting the `ctx.Done()` case in `waitForPayload` and not pulling the actual payload out of the channel.

Reply:
Yes and no. The tests didn't run in parallel, so as long as things were working mostly correctly, there wouldn't be test interference. But like you said, in the situation where one test does something unexpected (like time out), the outcome of one test would affect others in a way that's difficult to debug. I was making an assumption, that we now know from the trace is incorrect, that starting up a new client/server every time was costing us and causing the timeout. But now that we know the root cause, I'm going to revert a lot of the commits to this PR before we merge it.

Reply:
This change has now been reverted.